src/main/java/bizbook/logic/commands/AddCommand.java

```diff
@@ -56,6 +56,9 @@ public CommandResult execute(Model model) throws CommandException {
             throw new CommandException(MESSAGE_DUPLICATE_PERSON);
         }
 
+        // Save current version of the address book
+        model.saveAddressBookVersion();
```

Comment:
This implementation involves a lot of command coupling, would it be better to extract out the model.saveAddressBookVersion() to the logicManager, just like save to storage? 
Then to ensure you only save commands that actually affect the db, you can add an equals check within the commit() to check if the new addressbook is different from the prev one.

Reply:
I can't move it to `LogicManager` since logic manager does not differentiate what commands edits the addressbook and will save after each command execution.

But I moved it to `ModelManager` instead since I believe that when the model gets updated it anyway it will save a copy of itself before making the change. This I feel should help resolve the coupling.